### PR TITLE
Feature/fixing validation report

### PR DESCRIPTION
**Pull request by damniko**
Added:
- Show Empty-button to hide managers with no shown validation tests
- Added a style for the Copy SQL-buttons
- Added tooltips where appropriate
- Added functionality to search for context ID

Bugfixes:
- Fixed bug where different validation report viewmodels were connected through the same "CollectionViewSource", and filtering affected all views
- Filtering is completely fixed and works as well as it did before adding the ability to view different executions
- Added a validation queue to ensure that if a validation test's manager is not yet registered in the system, it will be added at a later time when said manager exists

Others:
- Added a global exception-handling method in MainWindow.xaml.cs, which catches any unhandled exceptions that may occur (when not in debug-mode)

(Kasper: Læg mærke til hvor flot formatteret mit XAML er!)

### CODE REVIEW

File: DashboardFrontend/Controller.cs [hunk 1]
```diff
@@ -260,20 +261,42 @@ public void UpdateValidationReport()
...
+                var v = ValidationQueue.Peek();
+                if (Conversion.AllManagers.Find(m => m.Name.Contains(v.ManagerName) && v.Date < m.EndTime) is Manager mgr)
+                {
+                    mgr.Validations.Add(ValidationQueue.Dequeue());
+                    retryCount = 0;
+                    hasUpdatedData = true;
+                }
+                retryCount++;
+            }
+
```
Comment (mirakst) [inline DashboardFrontend/Controller.cs:279]: god tilføjelse!

File: DashboardFrontend/DetachedWindows/ValidationReportDetached.xaml [hunk 2]
```diff
@@ -25,286 +25,288 @@
...
+                <StackPanel Grid.Column="1"
+                            Orientation="Horizontal" 
+                            HorizontalAlignment="Right">
+                    <ToggleButton Style="{DynamicResource DefaultToggleButton}"
+                                  IsChecked="{Binding ShowEmpty,UpdateSourceTrigger=PropertyChanged}"
+                                  Margin="0,2,0,2">
+                        <StackPanel Orientation="Horizontal" Margin="6 0 6 0">
+                            <TextBlock Text="Show empty"
+                                       VerticalAlignment="Center"/>
+                        </StackPanel>
```
Comment (mirakst) [inline DashboardFrontend/DetachedWindows/ValidationReportDetached.xaml]: jeg er lidt xml-døv, men giver det mening med et 1-child stackpanel? kan du ikke bare smide padding på textblocken?

Comment (mirakst) [reply DashboardFrontend/DetachedWindows/ValidationReportDetached.xaml]: BTW så er det her en 6 l/r 0 t/b margin - det resultat kan opnås med Margin="6 0"

File: DashboardFrontend/DetachedWindows/ValidationReportDetached.xaml [hunk 3]
```diff
@@ -25,286 +25,288 @@
...
+                                   Margin="0 0 2 0"
+                                   MaxHeight="20"/>
+                            <TextBlock Text="{Binding SelectedExecution.FailedTotalCount,FallbackValue='N/A'}"
+                                       VerticalAlignment="Center"/>
+                        </StackPanel>
+                    </ToggleButton>
+                    <StackPanel Orientation="Horizontal" Margin="6 0 6 0">
+                        <TextBlock Text="{Binding SelectedExecution.TotalCount,FallbackValue='N/A',StringFormat='(of {0})'}"
+                                   VerticalAlignment="Center"
+                                   Foreground="Gray"/>
```
Comment (AndersMariegaard) [inline DashboardFrontend/DetachedWindows/ValidationReportDetached.xaml]: Der er også et et-child stackpanel hernede hvis det var

File: DashboardFrontend/DetachedWindows/ValidationReportDetached.xaml [hunk 4]
```diff
@@ -25,286 +25,288 @@
...
+                              Style="{DynamicResource ComboBoxExecutions}" 
+                              ItemsSource="{Binding Executions}" 
+                              SelectedItem="{Binding SelectedExecution, UpdateSourceTrigger=PropertyChanged, Mode=TwoWay}">
+                        <ComboBox.ItemTemplate>
+                            <DataTemplate>
+                                <StackPanel Orientation="Horizontal">
+                                    <TextBlock Text="{Binding Id, StringFormat='Execution {0}'}" 
+                                               VerticalAlignment="Center"
+                                               Foreground="{DynamicResource HeaderForegroundColor}"/>
+                                </StackPanel>
```
Comment (AndersMariegaard) [inline DashboardFrontend/DetachedWindows/ValidationReportDetached.xaml]: Og her

File: DashboardFrontend/MainWindow.xaml [hunk 5]
```diff
@@ -565,8 +570,9 @@
                                          Margin="2"
                                          Style="{DynamicResource ValidationsSearchBox}"
                                          Text="{Binding NameFilter, UpdateSourceTrigger=PropertyChanged}"
-                                         Foreground="White"
-                                         CaretBrush="White"/>
+                                         ToolTip="Search for managers by their name or context ID"
+                                         Foreground="{DynamicResource ForegroundColor}"
+                                         CaretBrush="{DynamicResource ForegroundColor}"/>
```
Comment (mirakst) [inline DashboardFrontend/MainWindow.xaml:571]: jaa

File: DashboardFrontend/MainWindow.xaml.cs [hunk 6]
```diff
@@ -1,26 +1,32 @@
...
-using DashboardFrontend.Charts;
-using LiveChartsCore.SkiaSharpView.WPF;
 
 namespace DashboardFrontend
 {
     public partial class MainWindow : Window
     {
         public MainWindow()
         {
+            this.Dispatcher.UnhandledException += OnDispatcherUnhandledException;
```
Comment (mirakst) [inline DashboardFrontend/MainWindow.xaml.cs:18]: sindssygt

**Review by AndersMariegaard — APPROVED** [review]

File: DashboardFrontend/ViewModels/ValidationReportViewModel.cs [hunk 7]
```diff
@@ -127,10 +137,12 @@ private void SetExecution(ExecutionObservable exec)
         {
             if (exec is not null)
             {
+                ExpandedManagerNames.Clear();
                 ManagerView = (CollectionView)CollectionViewSource.GetDefaultView(exec.Managers);
                 ManagerView.Filter = OnManagersFilter;
                 ManagerView.SortDescriptions.Add(new(nameof(ManagerObservable.FailedCount), ListSortDirection.Descending));
                 ManagerView.SortDescriptions.Add(new(nameof(ManagerObservable.DisabledCount), ListSortDirection.Descending));
+                ManagerView.SortDescriptions.Add(new(nameof(ManagerObservable.StartTime), ListSortDirection.Ascending));
```
Comment (mirakst) [inline DashboardFrontend/ViewModels/ValidationReportViewModel.cs:145]: 😍

File: DashboardFrontend/ViewModels/ValidationReportViewModel.cs [hunk 8]
```diff
@@ -142,33 +154,29 @@ private void SetExecution(ExecutionObservable exec)
         private bool OnManagersFilter(object item)
         {
             ManagerObservable mgr = (ManagerObservable)item;
-            return mgr.Name.Contains(NameFilter);
+            return (mgr.Name.Contains(NameFilter) || mgr.ContextId.ToString() == NameFilter) && (!mgr.ValidationView.IsEmpty || ShowEmpty);
```
Comment (mirakst) [inline DashboardFrontend/ViewModels/ValidationReportViewModel.cs:157]: 💪

**Review by mirakst — COMMENTED** [review]
Det er godt! 
Havde lige et spørgsmål eller to som du gerne må kigge på.
Ellers er det bare små udtryk af taknemmelighed

**Review by KasperHenningsen — APPROVED** [review]

**Pull request merged**